tests/transforms/test_add_noise.py

```diff
@@ -0,0 +1,458 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the insert transform.
+"""
+from functools import partial
+
+import numpy as np
+import pytest
+
+import pennylane as qml
+from pennylane.measurements import Expectation
+from pennylane.tape import QuantumScript
+from pennylane.transforms.add_noise import add_noise
+
+# pylint:disable = no-member
+
+
+class TestAddNoise:
+    """Tests for the insert function using input tapes"""
+
+    with qml.queuing.AnnotatedQueue() as q_tape:
+        qml.RX(0.9, wires=0)
+        qml.RY(0.4, wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.RY(0.5, wires=0)
+        qml.RX(0.6, wires=1)
+        qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    tape = QuantumScript.from_queue(q_tape)
+
+    with qml.queuing.AnnotatedQueue() as q_tape_with_prep:
+        qml.StatePrep([1, 0], wires=0)
```

Comment:
Why did you have to separate the two cases?

Reply:
One is with `StatePrep` and one is without. Some other transforms also test them as separate cases so I've followed them.